payment-method-messaging/src/main/java/com/stripe/android/paymentmethodmessaging/view/injection/PaymentMethodMessagingModule.kt

```diff
@@ -0,0 +1,90 @@
+package com.stripe.android.paymentmethodmessaging.view.injection
+
+import android.app.Application
+import android.content.Context
+import android.graphics.Bitmap
+import android.text.style.ImageSpan
+import androidx.core.text.HtmlCompat
+import com.stripe.android.BuildConfig
+import com.stripe.android.core.injection.ENABLE_LOGGING
+import com.stripe.android.core.injection.PUBLISHABLE_KEY
+import com.stripe.android.model.PaymentMethodMessage
+import com.stripe.android.paymentmethodmessaging.view.PaymentMethodMessagingData
+import com.stripe.android.paymentmethodmessaging.view.PaymentMethodMessagingView
+import com.stripe.android.payments.core.injection.PRODUCT_USAGE
+import com.stripe.android.uicore.image.StripeImageLoader
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.withContext
+import javax.inject.Named
+
+@Module
+internal object PaymentMethodMessagingModule {
+    @Provides
+    fun providesAppContext(application: Application): Context = application
+
+    @Provides
+    @Named(PUBLISHABLE_KEY)
+    fun providePublishableKey(
+        configuration: PaymentMethodMessagingView.Configuration
+    ): () -> String = { configuration.publishableKey }
+
+    @Provides
+    @Named(PRODUCT_USAGE)
+    fun providesProductUsage(): Set<String> = emptySet()
+
+    @Provides
+    @Named(ENABLE_LOGGING)
+    fun providesEnableLogging(): Boolean = BuildConfig.DEBUG
+
+    @Provides
+    fun providesStripeImageLoader(
+        application: Application
+    ): StripeImageLoader = StripeImageLoader(application)
+
+    @Provides
+    fun providesMapper(
+        scope: CoroutineScope,
```

Comment:
Injecting a scope here doesn't follow structured concurrency best practices. If the caller cancels, we want to cancel our own work.

In practice, it's going to happen that way, but it's better to design by having the caller pass in the scope. This could be achieved by changing the signature to `(CoroutineScope, PaymentMethodMessage) -> Deferred<PaymentMethodMessagingData>`.